src/perfetto_cmd/perfetto_cmd.cc

```diff
@@ -925,6 +981,22 @@ int PerfettoCmd::ConnectToServiceRunAndMaybeNotify() {
 }
 
 int PerfettoCmd::ConnectToServiceAndRun() {
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
+  if (upload_after_reboot_flag_) {
+    std::string uploader_ready_prop =
+        base::GetAndroidProp("perfetto.persistent.uploader.status");
```

Comment:
I'm a bit lukewarm on doing all these pedantic checks? What I worry about is that I am not sure you can read init.svc selinux-wise

I'd just keep it easy: if you ask for --upload-after-reboot, just do it.

The uploader status shouldn't matter as WaitForService will wait for it ... forever